src/lib/y2storage/proposal/autoinst_devices_planner.rb

```diff
@@ -115,6 +121,37 @@ def planned_for_disk(disk, drive)
         result
       end
 
+      # Returns an array of planned Xen partitions according to a <drive>
+      # section which groups virtual partitions with a similar name (e.g. a
+      # "/dev/xvda" section describing "/dev/xvda1" and "/dev/xvda2").
+      #
+      # @param drive [AutoinstProfile::DriveSection] drive section describing
+      #   a set of stray block devices (Xen virtual partitions)
+      # @return [Array<Planned::StrayBlkDevice>] List of planned devicess
+      def planned_for_stray_devices(drive)
+        result = []
+        drive.partitions.each_with_index do |section|
+          name = drive.device + section.partition_nr.to_s
```

Comment:
What happens if the `partition_nr` is not specified? We have several options:

* Use the `index` number (so you might need the `each_with_index` after all).
* Report that the value is missing and ignore the section.